In [1]:
load('../../F21-ETALA/etala_utilities.sage')

##### F21ETSMP Group Assignment 10
# Height Study

In a study of women’s heights, we obtain the following measurements:  

In [3]:
data = [1.66,  1.69,  1.50,  1.80,  1.68,  1.64,  1.65,  1.70,  1.72,  1.67,  1.69,  1.68,  1.20,  1.90,  1.2]

In the population, the heights of woman is assumed to be Gaussian (normal) distributed with a mean of 1.68m and a standard deviation of 0.2m.  
Assume that you only test on the mean.

## 1. 
Estimate the mean of the population sample.

Since the population is assumed to be Gaussian distributed, the mean is estimated as  
$ \hat \mu = \frac {1}{n} \sum_{i=1}^n X_i \sim \mathcal N \left(\mu, \sigma^2\right)$

In [8]:
data_est_mean = sum(data)/len(data); show_var()

\text{data_est_mean} = 1.62533333333333

## 2.
Formulate the NULL hypothesis to test whether the sample has the sample mean as the same mean as the rest of the population.

The null hypothesis for this test would be  
$\text{$H_0$:}\;\hat\mu = \mu$

## 3.  
Formulate the alternative hypothesis to the NULL hypothesis.

The alternative hypothesis is then  
$\text{$H_1$:}\;\hat\mu \ne \mu$

## 4.
Calculate the test statistics z.

Since the mean and variance of the population is known, the Z-score can be calculated. The Formula is as follows:  
$Z = \dfrac{\hat\mu - \mu}{\sigma / \sqrt n}$

In [10]:
pop_std_dev = 0.2
pop_mean = 1.68
Z = (data_est_mean-pop_mean)/(pop_std_dev/sqrt(len(data))); show_var()

\text{Z} = -0.273333333333333 \, \sqrt{15} \sim -1.06

## 5.
Find the p-value based on a Gaussian pdf.

Since the mean could deviate in either direction, a two tailed p-value is calculated.  
$ p = 2 \cdot min\left\{Pr(X \le x \vert H_0), Pr(X \ge x \vert H_0)\right\}$  
Now given that the test statistic is normalized, the probability $Pr(X \le x \vert H_0) = \Phi(Z)$

In [21]:
from scipy.stats import norm
p = 2 * min([norm.cdf(float(Z)), 1 - norm.cdf(float(Z))]); show_var()

\text{p} = 0.28977494865740705 \sim 0.290

## 6.
With a significance level of  $\alpha = 0.05$, can we reject the NULL hypothesis?

Since the p value is larger than the significance level, __the NULL hypothesis cannot be rejected__.

## 7.
If we changes the significance level to α=0.1, what would that imply?

__This would still not allow rejection of the NULL hypothesis.__ Furthermore, given that the p value indicates that there's a 29% chance of the given observation under $H_0$, the NULL hypothesis will not be rejected under any other significance level we might be likely to use.

## 8.
